Adding in generic update instructions and fixing some links. (#99)
timescale/docs: timescaledb/how-to-guides/update-timescaledb/update-timescaledb.md

# Updating major versions of TimescaleDB [](update)

Use these instructions to update TimescaleDB within the same major release

> version (eg. TimescaleDB 2.1->2.2 or 1.7->1.7.4). If you need to upgrade between

Reviewer: ```suggestion
version (for example, from TimescaleDB 2.1 to 2.2, or from 1.7 to 1.7.4). If you need to upgrade between
```

> TimescaleDB 1.x and 2.x, please see our [separate upgrade document][update-tsdb-2] 

Reviewer: ```suggestion
TimescaleDB 1.x and 2.x, see our [separate upgrade document][update-tsdb-2] 
```

> for more details instructions.

Reviewer: ```suggestion
for detailed instructions.
```

### TimescaleDB release compatibility

TimescaleDB currently supports the following PostgreSQL releases. If you are not currently running
a compatible release, please upgrade before updating TimescaleDB.

 TimescaleDB Release |   Supported PostgreSQL Release
 --------------------|-------------------------------
 1.7                 | 9.6, 10, 11, 12
 2.0                 | 11, 12

>  2.1+                | 11, 12, 13

Reviewer 2: Do we want to proactively note that 2.4+ is proposed to drop support of Postgres 11?  Otherwise, might be easy to not know this, as only in release notes.

> If you need to upgrade PostgreSQL first, please see [our documentation][upgrade-pg].

Reviewer: ```suggestion
If you need to upgrade PostgreSQL first, see [our documentation][upgrade-pg].
```

> If you need to upgrade PostgreSQL first, see [our documentation][upgrade-pg].

Reviewer 2: Explicitly note that if you want to upgrade both TimescaleDB & major PostreSQL versions, this should be done as two separate processes.  E.g., do a full TimescaleDB upgrade, then to PG upgrade.  Or, do full PG upgrade then TimescaleDB upgrade.  Don't try to combine them.

### Update TimescaleDB

Software upgrades use PostgreSQL's `ALTER EXTENSION` support to update to the
latest version. TimescaleDB supports having different extension
versions on different databases within the same PostgreSQL instance. This
allows you to update extensions independently on different databases. The

> upgrade process involves three-steps:

Reviewer: ```suggestion
upgrade process involves three steps:
```

> 1. We recommend that you perform a [backup][] of your database via `pg_dump`.

Reviewer: ```suggestion
1. Perform a [backup][] of your database via `pg_dump`.
```

Adding the "we recommend" weakens the statement here. We definitely want them to do this.
